firefox_ui_harness/runners/update.py

```diff
@@ -0,0 +1,76 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import shutil
+import tempfile
+
+import mozfile
+import mozinstall
+
+from base import FirefoxUITestRunner
+from ..testcases import UpdateTestCase
+
+
+DEFAULT_PREFS = {
+    'app.update.log': True,
+}
+
+
+class UpdateTestRunner(FirefoxUITestRunner):
+
+    def __init__(self, **kwargs):
+        FirefoxUITestRunner.__init__(self, **kwargs)
+
+        self.original_bin = self.bin
+
+        self.prefs.update(DEFAULT_PREFS)
+
+        # In case of overriding the update URL, set the appropriate preference
+        override_url = kwargs.pop('update_override_url', None)
+        if override_url:
+            self.prefs.update({'app.update.url.override': override_url})
+
+        self.run_direct_update = not kwargs.pop('update_fallback_only', False)
+        self.run_fallback_update = not kwargs.pop('update_direct_only', False)
+
+        self.test_handlers = [UpdateTestCase]
+
+    def duplicate_application(self, application_folder):
+        """Creates a copy of the specified binary."""
+        target_folder = tempfile.mkdtemp('.binary-update-tests')
+
+        self.logger.info('Creating a copy of the application at "%s".' % target_folder)
+        mozfile.remove(target_folder)
+        shutil.copytree(application_folder, target_folder)
+
+        return target_folder
+
+    def run_tests(self, tests):
+        source_folder = self.get_application_folder(self.original_bin)
+
+        def _run_tests(update_type):
+            target_folder = None
+
+            try:
+                target_folder = self.duplicate_application(source_folder)
+                self.bin = mozinstall.get_binary(target_folder, 'Firefox')
+
+                self.type = update_type
+                FirefoxUITestRunner.run_tests(self, tests)
+
+            except Exception:
+                # TODO: Store exception and re-raise after all tests have been run
+                self.logger.exception('Failure during execution of the update test.')
+
+            finally:
+                self.logger.info('Removing copy of the application at "%s"' % target_folder)
+                mozfile.remove(target_folder)
+
+        # Run direct update tests if wanted
+        if self.run_direct_update:
+            _run_tests(update_type='direct')
+
+        # Run fallback update tests if wanted
+        if self.run_fallback_update:
+            _run_tests(update_type='fallback')
```

Comment:
Maybe you are confused because the CLI options are named `--update-directly-only` instead of `--update-skip-fallback`? If not lets have a Vidyo chat to get that clarified.